src/main/java/seedu/address/model/pet/Species.java

```diff
@@ -12,7 +12,7 @@ public class Species implements Comparable<Species> {
      * @param value The string representation of the species.
      */
     public Species(String value) {
-        if (value == null) {
+        if (value == null || !value.matches("[a-zA-Z]+")) {
```

Comment:
Does a string with whitespaces match this regex?

Reply:
this one no

Reply:
let me change it